hyphalnet/CPTAC_threshold_1.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Jun 30 10:37:37 2020
+
+@author: linth
+"""
+import pandas as pd
+from scipy import stats
+data = pd.read_csv('CPTAC.csv')
+data.info()
+data.isnull().sum()
+data.head()
+# Creating protein dictionary
+aml_sample = data['AML sample']
+gene = data['Gene']
+log_fold = data['LogFoldChange']
+protein_dict = dict()
+n = len(aml_sample)
+for i in range(n):
+    protein_dict[aml_sample[i]]={gene[i]:log_fold[i]}
+protein_dict
+ #Calculating z score
+data['zscore'] = stats.zscore(data['LogFoldChange'])
+data.head()
+data['pvalue'] = stats.norm.cdf(data['zscore'])
+significant_genes = data[data['pvalue'] < 0.01]
+non_significant = data[data['pvalue'] > 0.01]
+

```

Comment:
What you can try instead is this:
```
meanVals=data.groupby('AML sample')['LogFoldChange'].mean()
stdVals =data.groupby('AML sample')['LogFoldChange'].std()
statDf = pd.DataFrame({'mean':meanVals,'std':stdVals}).reset_index()
fullDf = data.merge(statDf,on='AML sample')
##now we can compute the z score for each
fullDf['zscore'] = (newDf['LogFoldChange']-newDf['meanVals'])/newDf['std'])
##now we can take the z scores that fall above or below a certain threshold, e.g. 
newDf[abs(newDf['zscore']>5].groupby('AML Sample').count()
```

Reply:
I chose an absolulte z-score threshold because it's easier than selecting the p-value, but this is sufficient for now.